File: crud_api/app/schemas/service.py
from pydantic import BaseModel

from .config import config_dict


class ServiceBasicOut(BaseModel):
    """Service output without pricing - for appointment displays."""
    id: int
    name: str

    model_config = config_dict


class ServiceOut(ServiceBasicOut):
    """Service output with pricing - for booking form constraints."""
    amount_minor: int
    currency_code: int

File: tg_mini_app/src/types/Service.ts
import z from 'zod';

/** Service without pricing - for appointment displays */
export const ServiceBasicIn = z.object({
  id: z.number().int(),
  name: z.string(),
});

/** Service with pricing - for booking form constraints */
export const ServiceIn = ServiceBasicIn.extend({
  amountMinor: z.number().int(),
  currencyCode: z.number().int(),
});

export type ServiceBasic = z.infer<typeof ServiceBasicIn>;
export type Service = z.infer<typeof ServiceIn>;
